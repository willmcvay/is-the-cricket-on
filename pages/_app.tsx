import App, { Container, AppComponentProps } from 'next/app'
import React from 'react'
import { ApolloProvider } from 'react-apollo'
import withApollo from '../utils/with-apollo'
import ApolloClient from 'apollo-boost'

type Props = AppComponentProps<any> & {
  apollo: ApolloClient<any>
}

class MyApp extends App<Props> {
  render() {
    const { Component, pageProps, apollo } = this.props
    return (
      <Container>
        <ApolloProvider client={apollo}>
          <Component {...pageProps} />
        </ApolloProvider>
      </Container>
    )
  }
}

export default withApollo(MyApp as any)
