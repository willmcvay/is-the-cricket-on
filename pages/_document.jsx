// @flow
import React from 'react'
import Head from 'next/head'

const Document = () => (
	<Head>
		<title>Is the Cricket on?</title>
	</Head>
)

export default Document
