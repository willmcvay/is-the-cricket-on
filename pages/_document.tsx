import React from 'react'
import Document, {
  Head,
  Main,
  NextScript,
  NextDocumentContext,
  RenderPageResponse
} from 'next/document'
import { ServerStyleSheet } from 'styled-components'
import globalStyles from '../styles/base/global/global'

interface DocumentProps extends RenderPageResponse {
  styleTags: React.ReactElement<{}>[]
}

class AppDocument extends Document<DocumentProps> {
  static getInitialProps({ renderPage }: NextDocumentContext) {
    const sheet = new ServerStyleSheet()
    const page = renderPage(App => props =>
      sheet.collectStyles(<App {...props} />) as React.ReactElement<any>
    )
    const styleTags = sheet.getStyleElement()
    return { ...page, styleTags }
  }

  render() {
    return (
      <html>
        <Head>
          <meta
            name="viewport"
            content="width=device-width, initial-scale=1.0, maximum-scale=1.0"
          />
          <title>Is the Cricket on?</title>
          <link href="https://fonts.googleapis.com/css?family=Raleway" rel="stylesheet" />
          {this.props.styleTags}
          {globalStyles()}
        </Head>
        <Main />
        <NextScript />
      </html>
    )
  }
}

export default AppDocument
