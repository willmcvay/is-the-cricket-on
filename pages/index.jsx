// @flow
import React from 'react'
import Head from 'next/head'

const Index = () => (
	<div>
		<Head>
			<title>Is the Cricket on?</title>
		</Head>
		<div>Is the cricket on?</div>
	</div>
)

export default Index
