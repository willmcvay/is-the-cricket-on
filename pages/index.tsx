import * as React from 'react'
import { graphql, ChildDataProps } from 'react-apollo'
import upcomingMatches from '../shared/graphql/queries/upcoming-matches.graphql'

export const HomePage = (props: ChildDataProps) => {
  console.log('PROPS', props)

  return (
    <div>
      <div>Hello World</div>
    </div>
  )
}

// second param response
export default graphql<{}, {}>(upcomingMatches)(HomePage)
