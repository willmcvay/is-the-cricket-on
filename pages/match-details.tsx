import * as React from 'react'
import { ChildProps } from 'react-apollo'
import { Query } from '../shared/types/queries'
import { withRouter, WithRouterProps } from 'next/router'
import MatchDetailsComponent from '../components/queries/match-details'
import { QueryParams } from '../shared/types/api'

export type MatchDetailsProps = ChildProps<{}, Query> & WithRouterProps<QueryParams.MatchDetails>

export const MatchDetails = (props: MatchDetailsProps) => (
  <MatchDetailsComponent queryString={props.router.query as QueryParams.MatchDetails} />
)

export default withRouter(MatchDetails)
