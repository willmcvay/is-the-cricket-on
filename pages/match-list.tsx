import * as React from 'react'
import MatchListQuery from '../components/query-providers/match-list'
import { QueryParams } from '../shared/types/api'
import { withRouter } from 'next/router'
import { Core } from '../shared/types/core'

export type MatchListProps = Core.ApolloWrappedRouterProps<QueryParams.MatchList>

const MatchList = (props: MatchListProps) => (
  <MatchListQuery queryString={props.router.query as QueryParams.MatchList} />
)

export default withRouter(MatchList)
