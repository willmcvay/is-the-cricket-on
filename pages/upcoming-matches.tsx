import * as React from 'react'
import { graphql, ChildProps } from 'react-apollo'
import upcomingMatches from '../shared/graphql/queries/upcoming-matches.graphql'
import { Query } from 'shared/types/queries'

export const UpcomingMatches = (_props: ChildProps<{}, Query>) => {
  return (
    <div>
      <div>Upcoming Matches</div>
    </div>
  )
}

export default graphql<{}, Query, {}>(upcomingMatches)(UpcomingMatches)
