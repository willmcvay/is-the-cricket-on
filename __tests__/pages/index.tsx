import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { HomePage } from '../../pages/index'

describe('HomePage', function() {
  const renderComponent = (updatedProps = {}) => {
    const props = {
      ...updatedProps
    }

    this.HomePage = shallow(<HomePage {...props} />)
  }

  it('should match a snapshot', () => {
    renderComponent()
    expect(toJson(this.HomePage)).toMatchSnapshot()
  })
})
