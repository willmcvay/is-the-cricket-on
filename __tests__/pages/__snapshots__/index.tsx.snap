// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match a snapshot 1`] = `
<div>
  <div>
    Hello World
  </div>
</div>
`;
